src/dialect/mod.rs

```diff
@@ -27,7 +28,15 @@ pub use self::mysql::MySqlDialect;
 pub use self::postgresql::PostgreSqlDialect;
 pub use self::sqlite::SQLiteDialect;
 
-pub trait Dialect: Debug {
+/// Determine if the specified dialect matched to
+/// the parsed dialect , used for dialect spefic behaviour
```

Comment:
I think an example will be easier to understand here:

```
/// `dialect_of!(parser is SQLiteDialect |  GenericDialect)` evaluates
/// to `true` iff `parser.dialect` is one of the `Dialect`s specified.
```

Reply:
changed